test/functorch/test_vmap.py

```diff
@@ -4567,6 +4873,264 @@ def test_vmap_exhaustive(self, device, dtype, op):
+                ),
+                xfail(
+                    "lu_unpack",
+                    device_type="mps",
+                ),
+                xfail(
+                    "msort",
+                    device_type="mps",
+                ),
+                xfail(
+                    "mvlgamma",
+                    "mvlgamma_p_1",
+                    device_type="mps",
+                ),
+                xfail(
+                    "mvlgamma",
+                    "mvlgamma_p_3",
+                    device_type="mps",
+                ),
+                xfail(
+                    "mvlgamma",
+                    "mvlgamma_p_5",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nan_to_num",
+                    device_type="mps",
+                ),
+                xfail(
+                    "ne",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.binary_cross_entropy",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.celu",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.elu",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.glu",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.hardsigmoid",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.interpolate",
+                    "nearest",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.leaky_relu",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.mish",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.prelu",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.rms_norm",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.selu",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.silu",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.threshold",
+                    device_type="mps",
+                ),
+                xfail(
+                    "nn.functional.upsample_nearest",
+                    device_type="mps",
+                ),
+                xfailIf(
+                    "polar",
+                    lambda sample: MACOS_VERSION < 14,
+                    device_type="mps",
+                ),
```

Comment:
Hi @jbschlosser, 

I'm having issue with using `xfailIf`, would you mind helping with? 

Per this xfail condition, I'm on macOS 13 so the xfail is triggered. When the xfail is triggered, I hit the below assert where I'm not quite sure how to deal with it. I'm not in dynamo/compile environment just regular eager mode.

https://github.com/pytorch/pytorch/blob/47ad351ff3d23d2ba39e817e8f46315f8b861d81/torch/testing/_internal/opinfo/core.py#L1626-L1630